test/integration/select_test.lua

```diff
@@ -217,32 +217,32 @@ pgroup.test_negative_first = function(g)
     t.assert_equals(objects, helpers.get_objects_by_idxs(customers, {2, 3, 4}))
 
     -- id >= 2
-    -- first -2 after 5
+    -- first -4 after 5
     local conditions = {
         {'>=', 'id', 2},
     }
-    local first = -2
+    local first = -4
     local after = crud_utils.flatten(customers[5], g.space_format)
     local result, err = g.cluster.main_server.net_box:call(
        'crud.select', {'customers', conditions, {first=first, after=after}})
 
     t.assert_equals(err, nil)
     local objects = crud.unflatten_rows(result.rows, result.metadata)
-    t.assert_equals(objects, helpers.get_objects_by_idxs(customers, {3, 4}))
+    t.assert_equals(objects, helpers.get_objects_by_idxs(customers, {2, 3, 4}))
```

Comment:
There are 7 tuples satisfying conditions: 2, 3, 4, 5, 6, 7, 8. Why 4 last tuples after 5 is 2, 3, 4 and not 6, 7, 8? I don't see any REQ iterators in the request options.

Reply:
after = customers[5]
first = -4
condition = age >= 2

So, it's traverse from customers[5] in backward order because first < 0 (1, 2, 3, 4) and filter results with id >= 2 (2, 3, 4).

Reply:
Description of after + negative first is fixed in separate commit